ports/geckolib/wrapper.rs

```diff
@@ -470,14 +467,16 @@ impl<'le> TElement for GeckoElement<'le> {
         unsafe { GeckoNode::from_raw(self.element as *mut RawGeckoNode) }
     }
 
-    fn style_attribute(&self) -> &Option<PropertyDeclarationBlock> {
+    fn style_attribute(&self) -> Option<&Arc<PropertyDeclarationBlock>> {
         let declarations = unsafe { Gecko_GetServoDeclarationBlock(self.element) };
         if declarations.is_null() {
-            &NO_STYLE_ATTRIBUTE
+            None
         } else {
-            GeckoDeclarationBlock::with(declarations, |declarations| {
-                unsafe { transmute(&declarations.declarations) }
-            })
+            let opt_ptr = GeckoDeclarationBlock::with(declarations, |declarations| {
+                // Use a raw poointer to extend the lifetime
```

Comment:
Can you comment here why it's fine? (Basically, because we keep a strong reference to it during all the styling process?)

@upsuper, may you know about a better way of describing this? If so, I'd appreciate leaving a reply here so Simon can include it, or adding it in another patch.